nautiluszim/scraper.py

```diff
@@ -349,82 +349,80 @@ def test_archiveless_collection(self):
         self.load_collection()
 
         (
-            duplicate_file_names,
-            missing_files,
-            none_url_files,
-            all_file_names,
+            duplicate_filename,
+            missing_filenames,
+            all_uris,
         ) = self.test_files()
 
-        if not all_file_names:
+        if not all_uris:
             raise ValueError("Collection is emtpy:\n")
 
-        if none_url_files:
-            raise ValueError(
-                "File(s) referenced in collection are not urls:\n - "
-                + "\n - ".join(none_url_files)
-            )
+        for uri in all_uris:
+            if not uri.startswith("http"):
+                raise ValueError(
+                    f"File referenced in collection which are not urls:\n - {uri}\n "
+                )
 
-        self.__test_missing_files(missing_files)
-        self.__test_duplicate_file_names(duplicate_file_names)
+        self._ensure_no_missing_files(missing_filenames)
+        self._ensure_no_duplicate_filenames(duplicate_filename)
 
     def test_archive_collection(self):
         """Test the collection.json with the archive file"""
         self.load_collection()
         with zipfile.ZipFile(self.archive_path, "r") as zh:
             all_names = zh.namelist()
-        duplicate_file_names, missing_files, _, _ = self.test_files(all_names)
+        duplicate_filename, missing_filenames, _ = self.test_files(all_names)
 
-        self.__test_missing_files(missing_files)
-        self.__test_duplicate_file_names(duplicate_file_names)
+        self._ensure_no_missing_files(missing_filenames)
+        self._ensure_no_duplicate_filenames(duplicate_filename)
 
-    def __test_missing_files(self, files):
-        if files:
-            raise ValueError(
-                "File(s) referenced in collection but missing:\n - "
-                + "\n - ".join(files)
-            )
+    def _ensure_no_missing_files(self, files):
+        if not files:
+            return
+        raise ValueError(
+            "File(s) referenced in collection but missing:\n - " + "\n - ".join(files)
+        )
 
-    def __test_duplicate_file_names(self, files):
-        if files:
-            raise ValueError(
-                "Files in collection are duplicate:\n - " + "\n - ".join(files)
-            )
+    def _ensure_no_duplicate_filenames(self, files):
+        if not files:
+            return
+        raise ValueError(
+            "Files in collection are duplicate:\n - " + "\n - ".join(files)
+        )
 
     def test_files(
-        self, all_names: List[str] | None = None
-    ) -> Tuple[Set[str], List[str], List[str], List[str]]:
+        self, available_filenames: Optional[List[str]] = None
+    ) -> Tuple[Set[str], List[str], List[str]]:
```

Comment:
You should probably cast the set to list when returning to harmonize with the rest